lib/elixir/lib/range.ex

```diff
@@ -461,6 +461,69 @@ defmodule Range do
     end
   end
 
+  @doc """
+  Checks if all points of range2 are inside range1.
+
+  It returns true if all of the following conditions are met:
+
+   * the bounds of range2 are fully within range1
+   * the step size of range2 is a multiple of range1's step size
+   * The start of range2 is a member of range1
```

Comment:
I think those two rules are wrong for ranges of size 1 and 0. I think the conditions are:

1. range2 within range1
2. all points in range2 (if any) belongs to range1

Checking multiples is an optimization.